src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/SpecValidationUtility.cs

```diff
@@ -329,6 +329,20 @@ private static void ValidateProjectMetadata(PackageSpec spec, IEnumerable<string
 
                 throw RestoreSpecException.Create(message, files);
             }
+
+            // spec.name and spec.RestoreMetadata.ProjectName should be the same
+            if (string.Equals(spec.Name, spec.RestoreMetadata.ProjectName, StringComparison.Ordinal))
+            {
+                var message = string.Format(
+                    CultureInfo.CurrentCulture,
+                    Strings.NonMatchingProperties,
+                    nameof(spec.Name),
+                    spec.Name,
+                    nameof(spec.RestoreMetadata.ProjectName),
+                    spec.RestoreMetadata.ProjectName);
+
+                throw RestoreSpecException.Create(message, files);
```

Comment:
Just curious to know where does this exception gets thrown n handled?

Reply:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs#L60

Reply:
🥇